src/OrchardCore/Orchard.Environment.Shell/Builders/ShellContainerFactory.cs

```diff
@@ -87,16 +87,18 @@ public IServiceProvider CreateContainer(ShellSettings settings, ShellBlueprint b
                 var feature = blueprint.Dependencies.FirstOrDefault(x => x.Key == startup.GetType()).Value.FeatureInfo;
 
                 ServiceCollection featureServiceCollection;
-                ServiceCollection startupServices = new ServiceCollection();
```

Comment:
I've added more details in a comment below. The issue is that we are creating this new startupServices collection for each call to a startup classes ConfigureServices method.